src/modules/dashboard/components/ActionsPage/StakingWidget/StakingWidget.tsx

```diff
@@ -0,0 +1,144 @@
+  rootHash: string;
+  colonyAddress: Address;
+  tokenDecimals: number;
+};
+
+const displayName = 'StakingWidget';
+
+const MSG = defineMessages({
+  title: {
+    id: 'dashboard.ActionsPage.StakingWidget.title',
+    defaultMessage: `Select the amount to back the motion`,
+  },
+  description: {
+    id: 'dashboard.ActionsPage.StakingWidget.description',
+    defaultMessage: `Stake is returned if the motion passes. If there is a dispute, and the motion loses, part or all of your stake will be lost.`,
+  },
+  stakeButton: {
+    id: 'dashboard.ActionsPage.StakingWidget.stakeButton',
+    defaultMessage: 'Stake',
+  },
+  objectButton: {
+    id: 'dashboard.ActionsPage.StakingWidget.objectButton',
+    defaultMessage: 'Object',
+  },
+});
+
+const StakingWidget = ({
+  motionId,
+  motionDomainId,
+  rootHash,
+  colonyAddress,
+  tokenDecimals,
+}: Props) => {
+  const { walletAddress } = useLoggedInUser();
+  const { data, loading } = useStakeMotionLimitsQuery({
+    variables: {
+      colonyAddress,
+      userAddress: walletAddress,
+      motionId,
+      rootHash,
+    },
+  });
+  const { data: userColonyAddressesData } = useUserColonyAddressesQuery({
+    variables: { address: walletAddress },
+  });
+  const validationSchema = yup.object().shape({
+    amount: yup.number().required(),
+  });
+
+  const transform = useCallback(
+    pipe(
+      mapPayload(({ amount }) => {
+        return {
+          amount: bigNumberify(amount).mul(bigNumberify(10).pow(tokenDecimals)),
+          userAddress: walletAddress,
+          rootHash,
+          colonyAddress,
+          motionId: bigNumberify(motionId),
+          motionDomainId,
+          vote: bigNumberify(1),
+        };
+      }),
+    ),
+    [walletAddress, colonyAddress, motionId],
+  );
+
+  if (loading || !data?.stakeMotionLimits || !userColonyAddressesData?.user) {
+    return null;
+  }
+
+  const {
+    user: { colonyAddresses },
+  } = userColonyAddressesData;
+
+  const isSubscribed = (colonyAddresses || []).includes(colonyAddress);
+  const { minStake, maxStake, requiredStake } = data.stakeMotionLimits;
+
+  return (
+    <ActionForm
+      initialValues={{
+        amount: minStake,
+      }}
+      validationSchema={validationSchema}
+      submit={ActionTypes.MOTION_STAKE}
+      error={ActionTypes.MOTION_STAKE_ERROR}
+      success={ActionTypes.MOTION_STAKE_SUCCESS}
+      transform={transform}
+    >
+      {({ values, isValid }) => (
+        <div className={styles.wrapper}>
+          <Heading text={MSG.title} className={styles.title} />
+          <p className={styles.description}>
+            <FormattedMessage {...MSG.description} />
+          </p>
+          <span className={styles.amount}>{values.amount}</span>
+          <div className={styles.sliderContainer}>
+            <Slider
+              name="amount"
+              value={values.amount}
+              min={minStake}
+              max={requiredStake}
+              limit={maxStake}
+            />
```

Comment:
The slider needs to show the token symbol next to the amount

![Screenshot from 2021-04-20 13-55-52](https://user-images.githubusercontent.com/1193222/115387879-e9609080-a1e3-11eb-96c2-9f3d72eaaac1.png)

Reply:
Step for the slider should be 0.25 not 1

Reply:
The slider should be:
- limit is the the totalstake - currentStake (only that difference)
- if the user doesn't have enough active tokens to cover the above difffence, then the new limit is the current users active tokens
- current value is the current staked so far _(the saga should update this, after you stake some more)_


Here's the example

Saga is already fully staked:
![Screenshot from 2021-04-20 14-29-12](https://user-images.githubusercontent.com/1193222/115388652-cd112380-a1e4-11eb-9da1-32252e3060c2.png)

But the slider still shows that I've staked 0, and allows me to stake the full requiredStake value
![Screenshot from 2021-04-20 14-28-32](https://user-images.githubusercontent.com/1193222/115388712-de5a3000-a1e4-11eb-851a-5d4d6cbeb72a.png)